plugins/org.eclipse.emf.ecore/src/org/eclipse/emf/ecore/plugin/EcorePlugin.java

```diff
@@ -668,7 +687,16 @@ public static final class Activator extends EMFPlugin.OSGiDelegatingBundleActiva
       @Override
       protected BundleActivator createBundle()
       {
-        return new Implementation();
+        if (IS_ECLIPSE_RUNNING)
+        {
+          return new Implementation();
+        }
+        else
+        {
+          // If we are running in OSGi but not in Eclipse, i.e., no extension registry.
+          //
+          return new PlainOSGiBundleActivator();
+        }
```

Comment:
This branch is taken if the OSGi runtime is available but the Equinox registry is not available.

That is a new path that was previously unsupported so this is effectively a new path that is added purely for you new purpose more purely OSGi-based runtime.

Reply:
Yes that’s intentional. An eclipse runtime generally doesn’t play nicely with dynamic additions and removals of bundles so doesn’t really seem (to me) applicable context/use-case.  Also, in such a context **without** dynamic addition and removal, the extension point can be used for this effect.

Reply:
Works for me!